File: 0001_0100/0065/0065.py
def diff(s, t):
    return sum(s[i] != t[i] for i in range(len(s)))
        
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = f.readline().strip()
    n = int(f.readline())
    t = [diff(s, x) for x in f.read().split()]
    m = min(t) if n > 0 else 0
    k = [str(i + 1) for i in range(n) if t[i] == m]
    q.write(str(len(k)) + "\n" + " ".join(k))

File: 0001_0100/0065/0065.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

func diff(a, b string) int {
	s := 0
	for i := range a {
		if a[i] != b[i] {
			s++
		}
	}
	return s
}

func main() {
	content, _ := ioutil.ReadFile("input.txt")
	q, _ := os.Create("output.txt")
	a := strings.Fields(string(content))
	n, _ := strconv.Atoi(a[1])
	var p []int
	m := len(a[0])
	for i := 2; i < n+2; i++ {
		d := diff(a[0], a[i])
		if d <= m {
			p = append(p, i-1)
			if d < m {
				m = d
				p = p[len(p)-1:]
			}
		}
	}
	fmt.Fprint(q, len(p), "\n", strings.Trim(fmt.Sprint(p), "[]"))
}
